Added otx18-device

tbrlek opened the pull request:
<!--
Thanks for submitting a pull request. Please fill the template below,
otherwise we will not be able to process this pull request.
-->

#### Summary
<!--
A short summary, referencing related issues:
Closes #0000, References #0000, etc.
-->

Added Onethinx module

#### Changes
<!-- What are the changes made in this pull request? -->

- ...
- ...

#### Notes for Reviewers
Onethinx module is a dual core module with integrated LoRa and LoRaWAN stacks (SX126x), integrated antenna and integrated secure element. Onethinx module is not an end device, but an easy to use module "like esp32 for LoRa/LoRaWAN". Integrated stack is located on the M0+ core, while the M4 core is free for user application. It communicates with the stack through our easy-to-use Onethinx LoRa/LoRaWAN API.

I was not able to use validate for some reason.

...

#### Release Notes
<!--
NOTE: This section is optional.

Any notes that we need to include in the Release Notes for the next release.
These notes are formatted as bullet points, written in past tense, and will be
combined with the labels of this Pull Request.

Always mention changes in API.
-->

- ...

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/TheThingsNetwork/lorawan-devices?pullRequest=231) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/TheThingsNetwork/lorawan-devices?pullRequest=231) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/TheThingsNetwork/lorawan-devices?pullRequest=231) it.</sub>

Jaime-Trinidad commented on vendor/index.yaml:
```diff
@@ -1683,3 +1683,8 @@ vendors:
     name: ElektronikSystem i Ume
     website: https://www.elsys.se/en/
+
+    - id: onethinx
```
@tbrlek onethinx already exist in index file (line 537), use that one.

Jaime-Trinidad reviewed (changes requested):
files looks fine, please attend comments

Jaime-Trinidad commented:
@tbrlek thanks for your submission please sign CLA

Jaime-Trinidad reviewed (approved):
changes look fine

Jaime-Trinidad commented on vendor/onethinx/otx18.yaml:
```diff
@@ -0,0 +1,116 @@
+firmwareVersions:
+  - # Firmware version
+    version: '1.0'
+    numeric: 1
```
please add the corresponding hardware version for this firmware in this case '18.02'

tbrlek replied:
I am not sure what to add under firmware version, or how to form it. Currently out latest version is 0x000000C1 (0xC1 for short, 193 in decimal), but we plan to release C2 in the comming days (194 decimal). All firmware versions that we release work on this single hardware version that we offer. difference is in minor bug fixes. Version naming is in hexadecimal.

Older clients use older versions. Should I name all of them? All version are built based on 1.0.2B.

tbrlek replied:
Disregard, I now see I need to add hardwareVersion to firmwareVersions

Jaime-Trinidad commented on vendor/onethinx/otx18.yaml:
```diff
@@ -0,0 +1,116 @@
+dataSheetURL: https://onethinx.github.io/Onethinx_Documentation/DoxyDocumentation/html/intro.html
+
+# Commercial information
+resellerURLs:
```
in this case are resellers, region can't be World Wide

tbrlek replied:
Currently modules can only be bought through us, and we are not "resellers" so I will delete this whole part, thanks!

Jaime-Trinidad reviewed (changes requested):
attend comments and run make validate fmt

Jaime-Trinidad commented:
@tbrlek please verify otx18-eu868-profile.yaml file it causes an error and run make validate fmt

tbrlek commented:
> @tbrlek please verify otx18-eu868-profile.yaml file it causes an error and run make validate fmt

I have seen errors in my part (Onethinx). I fixed these errors, but now I am getting an error elsewhere (in Comtac).

```
Error: Command failed: bin/validate-image ./vendor/comtac/lpn-cm4-foto.png
'bin' is not recognized as an internal or external command,
operable program or batch file.

    at ChildProcess.exithandler (child_process.js:319:12)
    at ChildProcess.emit (events.js:375:28)
    at maybeClose (internal/child_process.js:1055:16)
    at Socket.<anonymous> (internal/child_process.js:441:11)
    at Socket.emit (events.js:375:28)
    at Pipe.<anonymous> (net.js:675:12) {
  killed: false,
  code: 1,
  signal: null,
  cmd: 'bin/validate-image ./vendor/comtac/lpn-cm4-foto.png'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! lorawan-devices@1.0.0 validate: `node bin/validate.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the lorawan-devices@1.0.0 validate script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Tomislav\AppData\Roaming\npm-cache\_logs\2021-06-25T12_40_45_167Z-debug.log
make: *** [validate] Error 1
```

Jaime-Trinidad commented:
> > @tbrlek please verify otx18-eu868-profile.yaml file it causes an error and run make validate fmt
> 
> I have seen errors in my part (Onethinx). I fixed these errors, but now I am getting an error elsewhere (in Comtac).
> 
> ```
> Error: Command failed: bin/validate-image ./vendor/comtac/lpn-cm4-foto.png
> 'bin' is not recognized as an internal or external command,
> operable program or batch file.
> 
>     at ChildProcess.exithandler (child_process.js:319:12)
>     at ChildProcess.emit (events.js:375:28)
>     at maybeClose (internal/child_process.js:1055:16)
>     at Socket.<anonymous> (internal/child_process.js:441:11)
>     at Socket.emit (events.js:375:28)
>     at Pipe.<anonymous> (net.js:675:12) {
>   killed: false,
>   code: 1,
>   signal: null,
>   cmd: 'bin/validate-image ./vendor/comtac/lpn-cm4-foto.png'
> }
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! lorawan-devices@1.0.0 validate: `node bin/validate.js`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the lorawan-devices@1.0.0 validate script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     C:\Users\Tomislav\AppData\Roaming\npm-cache\_logs\2021-06-25T12_40_45_167Z-debug.log
> make: *** [validate] Error 1
> ```
> 
> Onethinx part:
> 
> ```
> onethinx: valid index
> onethinx: otx18: valid
> onethinx: otx18: EU863-870: profile onethinx/otx18-eu868-profile valid
> onethinx: otx18: US902-928: profile onethinx/otx18-us915-profile valid
> onethinx: otx18: otx18.jpg image has correct extension
> onethinx: otx18: otx18-package.jpg image has correct extension
> ```

@tbrlek update your branch, please decrease size of the image max is 2000x2000 and run make validate fmt, the error shouldn't present I run it locally without problems.

tbrlek commented:
> > > @tbrlek please verify otx18-eu868-profile.yaml file it causes an error and run make validate fmt
> > 
> > 
> > I have seen errors in my part (Onethinx). I fixed these errors, but now I am getting an error elsewhere (in Comtac).
> > ```
> > Error: Command failed: bin/validate-image ./vendor/comtac/lpn-cm4-foto.png
> > 'bin' is not recognized as an internal or external command,
> > operable program or batch file.
> > 
> >     at ChildProcess.exithandler (child_process.js:319:12)
> >     at ChildProcess.emit (events.js:375:28)
> >     at maybeClose (internal/child_process.js:1055:16)
> >     at Socket.<anonymous> (internal/child_process.js:441:11)
> >     at Socket.emit (events.js:375:28)
> >     at Pipe.<anonymous> (net.js:675:12) {
> >   killed: false,
> >   code: 1,
> >   signal: null,
> >   cmd: 'bin/validate-image ./vendor/comtac/lpn-cm4-foto.png'
> > }
> > npm ERR! code ELIFECYCLE
> > npm ERR! errno 1
> > npm ERR! lorawan-devices@1.0.0 validate: `node bin/validate.js`
> > npm ERR! Exit status 1
> > npm ERR!
> > npm ERR! Failed at the lorawan-devices@1.0.0 validate script.
> > npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> > 
> > npm ERR! A complete log of this run can be found in:
> > npm ERR!     C:\Users\Tomislav\AppData\Roaming\npm-cache\_logs\2021-06-25T12_40_45_167Z-debug.log
> > make: *** [validate] Error 1
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Onethinx part:
> > ```
> > onethinx: valid index
> > onethinx: otx18: valid
> > onethinx: otx18: EU863-870: profile onethinx/otx18-eu868-profile valid
> > onethinx: otx18: US902-928: profile onethinx/otx18-us915-profile valid
> > onethinx: otx18: otx18.jpg image has correct extension
> > onethinx: otx18: otx18-package.jpg image has correct extension
> > ```
> 
> @tbrlek update your branch, please decrease size of the image max is 2000x2000 and run make validate fmt, the error shouldn't present I run it locally without problems.

@Jaime-Trinidad 
1. Updated branch. 
2. Decreased size to below 2000 x 2000. 
3. Ran make validate fmt. it still shows error at the same point for me. If it works without problems at your end, that is okay for me.
```
Error: Command failed: bin/validate-image ./vendor/comtac/lpn-cm4-foto.png
'bin' is not recognized as an internal or external command,
operable program or batch file.
```

Jaime-Trinidad commented:
@tbrlek which node and npm versions are you using? `package-lock.json` must not change, please update versions and run it again.

tbrlek commented:
> @tbrlek which node and npm versions are you using? `package-lock.json` must not change, please update versions and run it again.

@Jaime-Trinidad 
The readme said I need these versions: Node.js version 14.x, npm version 6.x, Go version 1.15.x. These are the versions I have:
```
node -v -> v14.17.1

npm -v -> 6.14.13

go version -> go version go1.15.13 windows/amd64
```

After I run make deps, package-lock.json changes and I get a warning.
```
C:\Onethinx_Tomislav\CoreModuleFiles\Documentation\TTNv3_Device\lorawan-devices>make deps
npm install
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
audited 76 packages in 0.931s
```
Which version do I need to have?
